### PR TITLE
Enhancing File Control

### PR DESCRIPTION
**Pull request by infinitytmbots**
**Please describe the changes this PR makes and why it should be merged:**
Fixed the file from being written with "[object Object]" instead of the Objects.
Added a toggle option for this to avoid unnecessary conversions.
**Status**

- [ ] Code changes have been tested against the Discord API and the discord.js wrapper, or there are no code changes
- [ ] Documentation has been added/modified, or there is nothing to change (docs/mods.json)

**Semantic versioning classification:**

- [x] This PR changes DBM's interface (methods or parameters added to default methods)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by RigidStudios** [comment]
this won't work, familiarize yourself with DBM actions first. Also yeah definitely listen to ESLint screaming at you, it has a point :)

**Comment by infinitytmbots** [comment]
May I dm you on Discord?

**Comment by infinitytmbots** [comment]
And yep, ESLint errors have been fixed

File: actions/file_control_MOD.js [hunk 1]
```diff
@@ -143,6 +212,7 @@ class FileControl {
     const dirName = path.normalize(this.evalMessage(data.filepath, cache))
     const fileName = this.evalMessage(data.filename, cache)
     const line = parseInt(this.evalMessage(data.input2, cache))
+    const togglestat = this.evalMessage(data.togglestatus.checked, cache)
```
Comment (RigidStudios) [inline actions/file_control_MOD.js:215]: For `data.togglestatus` to be a valid data you need to add a string `togglestatus` into the `fields` array.

Comment (RigidStudios) [reply actions/file_control_MOD.js:215]: Additionally, this.evalMessage is used to evaluate strings, simply leave data.togglestatus, without .checked

**Comment by RigidStudios** [comment]
How about a Read File counterpart, to read into the proper JS format?

File: actions/file_control_MOD.js (hunk 1 above)
Comment (infinitytmbots) [reply actions/file_control_MOD.js]: .checked is required to catch the boolean

File: actions/file_control_MOD.js [hunk 2]
```diff
@@ -98,6 +160,13 @@ class FileControl {
         Input Text:<br>
         <textarea id="input" placeholder="Leave Blank For None." class="round" type="textarea" rows="10"></textarea><br><br><br><br><br><br><br><br><br><br><br>
       </div>
+      <div style="padding: 5px 10px 5px 5px">
+        <label class="switch">
+          <input onclick="if(checked){value='yes'}else{value='no'}"type="checkbox" id="togglestatus" value = 'no'>
+          <span  class="slider round"></span>
```
Comment (almeidx) [inline actions/file_control_MOD.js:166]:
```suggestion
          <input onclick="if(checked){value='yes'}else{value='no'}" type="checkbox" id="togglestatus" value='no'>
          <span class="slider round"></span>
```

**Review by almeidx — CHANGES_REQUESTED** [review]

**Comment by RigidStudios** [comment]
Add an `init` function for the HTML for the check-box to disappear when it's not available to an option:
```js
function hideToggle(element) {
  // Switch for the value of the other elements:
  // Hide:
  targetfield.classList.add('hidden')
  // Show:
  targetfield.classList.remove('hidden')
}

// To run on HTML initialization:
hideToggle(document.getElementById("togglestatus"))
```
Call it on change of filetask probably.

**Comment by infinitytmbots** [comment]
> Add an `init` function for the HTML for the check-box to disappear when it's not available to an option:
> 
> ```js
> function hideToggle(element) {
>   // Switch for the value of the other elements:
>   // Hide:
>   element.classList.add('hidden')
>   // Show:
>   element.classList.remove('hidden')
> }
> 
> // To run on HTML initialization:
> hideToggle(document.getElementById("togglestatus"))
> ```
> 
> Call it on change of filetask probably.

do you mean that the toogle should be visible if the user chooses json format?

**Comment by RigidStudios** [comment]
I mean that when insert at line is selected, you can hide the toggle, otherwise, show it.

**Comment by infinitytmbots** [comment]
Somethings wrong with the repository

**Pull request closed without merging**